lib/identity/index.js

```diff
@@ -200,4 +248,68 @@ export default class Identity extends CryptoConsumer {
       StorageConfig.fromObject(obj.storageConfig)
     )
   }
+
+  initiateRecovery(username, recoveryUrl) {
+    return this.initiateBrokerLogin(username, recoveryUrl)
+  }
+
+  completeRecovery(otp, noteId, recoveryUrl) {
+    return this.completeBrokerLogin(otp, noteId, recoveryUrl)
+  }
+
+  async initiateBrokerLogin(
+    username,
+    brokerUrl = `${this.config.apiUrl}/v1/identity/${this.config.realmId}/broker`
+  ) {
+    const payload = {
+      username,
+      action: 'initiate'
+    }
+    const request = await fetch(brokerUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(payload)
+    })
+    return validateResponseAsJSON(request)
+  }
+
+  async completeBrokerLogin(
+    authResponse,
+    noteId,
+    brokerUrl = `${this.config.apiUrl}/v1/identity/${this.config.realmId}/broker`
```

Comment:
can this be extracted into a class const?

Reply:
It relies on instance variables, but perhaps it could be extracted into a getter?

Reply:
going to leave this as a direct param for now.